contracts/contracts/liquidity/LiquidityReward.sol

```diff
@@ -0,0 +1,371 @@
+pragma solidity 0.5.11;
+
+import {
+    Initializable
+} from "@openzeppelin/upgrades/contracts/Initializable.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { StableMath } from "../utils/StableMath.sol";
+import { Governable } from "../governance/Governable.sol";
+
+//
+// LiquidityReward contract doles out reward for liquidity
+//   base off of Sushiswap's MasterChef: https://github.com/sushiswap/sushiswap/blob/master/contracts/MasterChef.sol
+//
+contract LiquidityReward is Initializable, Governable {
+    using SafeMath for uint256;
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+
+    // Info of each user.
+    struct UserInfo {
+        uint256 amount; // How many LP tokens the user has provided.
+        int256 rewardDebt; // Reward debt. See explanation below.
+        //
+        // We do some fancy math here. Basically, any point in time, the amount of Reward Tokens
+        // entitled to a user but is pending to be distributed is:
+        //
+        //   pending reward = (user.amount * pool.accRewardPerShare) - user.rewardDebt
+        //
+        // Whenever a user deposits or withdraws LP tokens to a pool. Here's what happens:
+        //   1. The pool's `accRewardPerShare` (and `lastRewardBlock`) gets updated.
+        //   2. User receives the pending reward sent to his/her address.
+        //   3. User's `amount` gets updated.
+        //   4. User's `rewardDebt` gets updated.
+    }
+
+    // Info of each pool.
+    struct PoolInfo {
+        IERC20 lpToken; // Address of LP token contract.
+        uint256 lastRewardBlock; // Last block number that Reward calculation occurs.
+        uint256 accRewardPerShare; // Accumulated Reward per share in reward precision. See below.
+    }
+
+    // The Reward token
+    IERC20 public reward;
+
+    // Reward tokens created per block in 1e18 precision.
+    uint256 public rewardPerBlock;
+
+    // Info on the LP.
+    PoolInfo public pool;
+    // total Reward debt, useful to calculate if we have enough to pay out all rewards
+    int256 public totalRewardDebt;
+    // Info of each user that stakes LP tokens.
+    mapping(address => UserInfo) public userInfo;
+    // The block number when Liquidity rewards ends.
+    uint256 public endBlock;
+
+    // for now assume it's OGN precision which is standard 1e18
+    uint256 public constant REWARD_PRECISION = 1e18;
+
+    event CampaignStarted(
+        uint256 rewardRate,
+        uint256 startBlock,
+        uint256 endBlock
+    );
+    event CampaignStopped(uint256 endBlock);
+    event Deposit(address indexed user, uint256 amount);
+    event Withdraw(address indexed user, uint256 amount);
+    event Claim(address indexed user, uint256 amount);
+    event EmergencyWithdraw(address indexed user, uint256 amount);
```

Comment:
I don't see the event EmergencyWithdraw used anywhere?

Reply:
Yes, lets get rid of this event.